File: app/src/main/java/com/kieran/winnipegbus/ActivityManager.java
package com.kieran.winnipegbus;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

public class ActivityManager {
    private static List<Activity> activities = new ArrayList<>();

    public static void refreshThemes() {
        for (Activity activity : activities)
            activity.recreate();
    }

    public static void addActivity(Activity activity) {
        activities.add(activity);
    }

    public static void removeActivity(Activity activity) {
        activities.remove(activity);
    }
}

File: app/src/main/java/com/kieran/winnipegbus/ActivityManager.kt
package com.kieran.winnipegbus

import android.app.Activity

import java.util.ArrayList

object ActivityManager {
    private val activities = ArrayList<Activity>()

    fun refreshThemes() {
        for (activity in activities)
            activity.recreate()
    }

    fun addActivity(activity: Activity) {
        activities.add(activity)
    }

    fun removeActivity(activity: Activity) {
        activities.remove(activity)
    }
}
